proto/asn1-pdu/x509_certificate_to_der.cc

```diff
@@ -43,6 +43,105 @@ DECLARE_ENCODE_FUNCTION(AlgorithmIdentifierSequence) {
   EncodeTagAndLength(kAsn1Sequence, der.size() - tag_len_pos, tag_len_pos, der);
 }
 
+DECLARE_ENCODE_FUNCTION(SubjectKeyIdentifier) {
+  Encode(val.key_identifier(), der);
+}
+
+DECLARE_ENCODE_FUNCTION(AuthorityKeyIdentifier) {
+  // Save the current size in |tag_len_pos| to place sequence tag and length
+  // after the value is encoded.
+  size_t tag_len_pos = der.size();
+
+  if (val.has_key_identifier()) {
+    size_t pos_of_tag = der.size();
+    Encode(val.key_identifier(), der);
+    // |key_identifier| is Context-specific with tag number 0 (RFC
+    // 5280, 4.2.1.1).
+    ReplaceTag(kAsn1ContextSpecific, pos_of_tag, der);
```

Comment:
```suggestion
    ReplaceTag(kAsn1ContextSpecific | 0x00, pos_of_tag, der);
```

This helps match the pattern of the subsequent lines and makes it "explicit" that there's an `| 0x00` tag identifier (even if it's a no-op that the compiler will remove, this is optimizing for the reader)